### PR TITLE
Fix PostgresTableCopyOperator

### PR DESCRIPTION
**Pull request by jjmurre**
See more explanation in the commit messages.

### CODE REVIEW

File: src/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+pytest_plugins = ["helpers_namespace"]
+
+
+@pytest.fixture
+def test_dag():
+    return DAG(
+        "test_dag",
+        default_args={"owner": "airflow", "start_date": datetime.datetime(2018, 1, 1)},
+        schedule_interval=datetime.timedelta(days=1),
```
Comment (gkoller) [inline src/tests/conftest.py:21]: Airflow uses `pendulum` for dates and times. We use it too in our Airflow DAGs, though inconsistently. I'd like us to standardize on it as well. Hence maybe you could consider using:
``python
import pendulum
from airflow.settings import TIMEZONE

...
"start_date": pendulum.datetime(2018, 1, 1, tz=TIMEZONE)},
schedule_interval=pendulum.duration(days=1)
```
This has the added benefit it uses an aware datetime like object as all our DAGs do as well.

Comment (jjmurre) [reply src/tests/conftest.py:21]: Good idea, always good to be consistent :-)

File: src/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        return_value=Connection(
+            host="localhost",
+            conn_type="postgres",
+            login=postgres_credentials.username,
+            password=postgres_credentials.password,
+            port=postgres.ports["5432/tcp"][0],
+        ),
+    )
+    hook = request.module.PostgresHook()
+    conn = hook.get_conn()
```
Comment (gkoller) [inline src/tests/conftest.py:89]: I have read the comment that `close()` might not be necessary. However as you still close the `conn` explicitly you might as well use: `with closing(hook.get_conn()) as conn:` That way the `conn.close()` will be called in the presence of exceptions.

File: src/tests/conftest.py (hunk 1 above)
Comment (jjmurre) [reply src/tests/conftest.py:21]: I will also remove the hardcoded date and use `pendulum.yesterday`.

Comment (gkoller) [reply src/tests/conftest.py:21]: Excellent!

Though with a `tz` right? As in `pendulum.yesterday(tz=TIMEZONE)`

File: src/tests/conftest.py (hunk 2 above)
Comment (jjmurre) [reply src/tests/conftest.py:89]: Yeah. I tried several approaches and thought that yielding from the `closing` context mgr. from a fixture does not work. But, it does. So, I'll change it and also get the cursor from a context mgr.
Initially I did not realize that the container fixture has function scope, so is postgres is re-initialized anyway.

File: src/tests/conftest.py (hunk 1 above)
Comment (jjmurre) [reply src/tests/conftest.py:21]: Yes.

File: src/requirements.in [hunk 3]
```diff
@@ -42,3 +42,8 @@ pendulum == 2.1.2
 paramiko==2.7.2
 pysftp==0.2.9
 sshtunnel==0.1.4
+pytest==6.2.5
+pytest-helpers-namespace==2021.04.29
+pytest-env==0.6.2
+pytest-mock==3.6.1
+pytest-docker-tool==3.1.0
```
Comment (vdboor) [inline src/requirements.in:49]: not pytest-docker-tool**s** ?

File: src/tests/conftest.py (hunk 2 above)
Comment (vdboor) [reply src/tests/conftest.py:89]: Yeah, context managers and yield work fine together. It's almost magic.

File: src/requirements.in (hunk 3 above)
Comment (jjmurre) [reply src/requirements.in:49]: Yep, good catch!

**Comment by gkoller** [comment]
I'm very happy that we now finally have a first test for Airflow. I'm struggling a bit with the Docker requirement though. Should work like a breeze under Linux, but is much slower on the Mac (runs in an automatically provisioned VM) and is a bit of a pain on FreeBSD (again, need to use a VM, but provision it manually).

I assume you could run these tests against `AIRFLOW_CONN_POSTGRES_DEFAULT` with the following setup:
- create `test_...` schema (session scope)
  - create connection (module scope)
    - explicitly start transaction (function scope)
      - run test function
    - rollback transaction(function scope)
  - close connection
- drop `test_...` schema 




Thought that's probably too much work to change right now.

File: src/tests/conftest.py (hunk 2 above)
Comment (jjmurre) [reply src/tests/conftest.py:89]: Yeah, same goes for the connection, I could remove `closing()` for both connection and cursor.

**Comment by jjmurre** [comment]
> I'm very happy that we now finally have a first test for Airflow. I'm struggling a bit with the Docker requirement though. Should work like a breeze under Linux, but is much slower on the Mac (runs in an automatically provisioned VM) and is a bit of a pain on FreeBSD (again, need to use a VM, but provision it manually).
> 
> I assume you could run these tests against `AIRFLOW_CONN_POSTGRES_DEFAULT` with the following setup:
> 
> * create `test_...` schema (session scope)
>   
>   * create connection (module scope)
>     
>     * explicitly start transaction (function scope)
>       
>       * run test function
>     * rollback transaction(function scope)
>   * close connection
> * drop `test_...` schema
> 
> Thought that's probably too much work to change right now.

Actually, I did not realize this is harder on Mac/FreeBSD. The docker approach is a bit slow, so it sure is worthwhile to investigate if a regular test database could work.

**Pull request merged**